src/planner/binder/expression/bind_columnref_expression.cpp

```diff
@@ -34,22 +34,34 @@ unique_ptr<ParsedExpression> ExpressionBinder::QualifyColumnName(const string &c
 			return move(coalesce);
 		}
 	}
+
 	// no table name: find a binding that contains this
+
+	if (lambda_bindings) {
+		for (idx_t i = 0; i < lambda_bindings->size(); i++) {
+			if ((*lambda_bindings)[i].HasMatchingBinding(column_name)) {
+				// first priority are lambda bindings TODO: throw a warning when this name conflicts
```

Comment:
I am not sure, I saw that it was left un-implemented for the macros so I assumed that it is not necessary @lnkuiper ?

Reply:
Huh, good point. For macro's this can conflict when creating a macro like this:
```sql
create table test as select range i from range(3);
create macro my_macro(i) as (select i in (select i from test)); -- ambiguous 'i', macro parameter takes prio
select my_macro(5); -- always returns true, even though test(i) does not contain 5
```

For clarity, I suppose it's better for both macro's and lambda's to throw an error when the `column_name` that is found in a macro/lambda binding can also be found in the regular bindings using `binder.bind_context.GetMatchingBinding(column_name);`, rather than giving priority silently.

Reply:
I changed this now, it should throw an error whenever there is ambiguity in the naming.